src/pages/Results/index.js

```diff
@@ -143,12 +144,20 @@ class Results extends Component {
   favoriteRecord = (event, value, result) => {
     // Stop propagation from going to the next page
     event.stopPropagation();
-    const { dispatch } = this.props;
-    // dispatch an action to favorite result
-    dispatch({
-      type: 'user/favoriteResult',
-      payload: result,
-    });
+    const {
+      auth: { username },
+    } = this.props;
+
+    if (username) {
+      const { dispatch } = this.props;
+      // dispatch an action to favorite controller
+      dispatch({
+        type: 'user/favoriteController',
+        payload: result,
+      });
+    } else {
+      console.log('Not logged on');
```

Comment:
Ideally, we would like to show a modal with a message that says "Register or Login into pbench dashboard" and a link redirecting to the login page. @anishaswain If there already exists one in our user auth PR, we can reuse that component?

Reply:
@aquibbaig we do not have any modal component redirecting to login page yet

Reply:
@anishaswain We can make one and reuse the same for the login component